src/Illuminate/Auth/Access/Grant.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Illuminate\Auth\Access;
+
+class Grant
+{
+    /**
+     * Indicates whether the request was granted.
+     *
+     * @var bool
+     */
+    protected $allowed;
+
+    /**
+     * The reason the request was granted/denied.
+     *
+     * @var string|null
+     */
+    protected $reason;
+
+    /**
+     * Create a new access grant instance.
+     *
+     * @return void
+     */
+    protected function __construct()
+    {
+        //
```

Comment:
Hmm, let's say non-experts then. I recon many Laravel users will have never come across code like this before.